File: public/js/mmm/rest/roles.js
angular.module('mmm.rest.roles', ['ngResource',
                                  'util.restutils'])

.factory('Roles', ['$resource', 'restUtils', function($resource, restUtils) {

  var Roles = $resource(
    '/api/admin/roles'
  );

  restUtils.attachInOut(Roles);

  return Roles;

}]); // 'factory'

File: app/controllers/mmm/admin/Roles.scala
package controllers.mmm.admin

import play.api.mvc._
import securesocial.core.SecureSocial
import util.JsonRestSuccess

object Roles
  extends Controller
  with SecureSocial {

  import util.MmmActions._

  implicit val implicitMmmRoleFormat = util.MmmJsonifier.MmmRoleFormat

  def query = MmmAuthAction("Admin") {
    Ok(JsonRestSuccess(db.Role.query()))
  }

}
